deps/wrapper.cc

```diff
@@ -344,11 +344,15 @@ JLCXX_MODULE define_julia_module(jlcxx::Module& mod)
     mod.method("get", &get);
     mod.method("_submit_task", &_submit_task);
 
-    // mod.add_type<RayObject>("RayObject")
-    //     .constructor<const std::shared_ptr<Buffer>,
-    //                  const std::shared_ptr<Buffer>,
-    //                  const std::vector<rpc::ObjectReference>,
-    //                  bool>();
+    // mod.add_type<rpc::ObjectReference>("ObjectReference");
+    mod.add_type<rpc::ObjectReference>("ObjectReference");
+
+    mod.add_type<RayObject>("RayObject")
+        .constructor<const std::shared_ptr<Buffer>,
+                     const std::shared_ptr<Buffer>,
+                     const std::vector<rpc::ObjectReference>,
+                     bool>()
```

Comment:
runtime error for 
```
julia> using ray_core_worker_julia_jll
[ Info: Precompiling ray_core_worker_julia_jll [c348cde4-7f22-4730-83d8-6959fb7a17ba]
C++ exception while wrapping module ray_core_worker_julia_jll: No appropriate factory for type NSt3__16vectorIN3ray3rpc15ObjectReferenceENS_9allocatorIS3_EEEE
ERROR: LoadError: No appropriate factory for type NSt3__16vectorIN3ray3rpc15ObjectReferenceENS_9allocatorIS3_EEEE
```

Reply:
I can add a workaround like this but I don't feel it should be necessary...
```
mod.method("RayObject", [] (
    const std::shared_ptr<Buffer> data, 
    const std::shared_ptr<Buffer> metadata, 
    rpc::ObjectReference obj_ref, 
    bool copy_data = false) {
    return RayObject(data, metadata, {obj_ref}, copy_data);
});
```

Reply:
It looks like you can add support for `std::vector`s of your own type by adding `jlcxx::stl::apply_stl<rpc::ObjectReference>(mod)` to the module.

https://github.com/JuliaInterop/CxxWrap.jl#stl-support

but now I'm getting a method error about converting a `SharedPtrAllocated{Buffer}`
```julia

julia> RayObject(data_buffer, meta_buffer, StdVector([nested_refs]), false)
ERROR: MethodError: no method matching __cxxwrap_smartptr_construct_from_other(::Type{CxxWrap.StdLib.SharedPtrDereferenced{Buffer}}, ::CxxWrap.CxxWrapCore.CxxRef{CxxWrap.StdLib.SharedPtr{Buffer}})

Closest candidates are:
  __cxxwrap_smartptr_construct_from_other(::Type{<:CxxWrap.CxxWrapCore.SmartPointer{T}}, ::CxxWrap.CxxWrapCore.SmartPointer{T}) where T
   @ CxxWrap ~/.julia/packages/CxxWrap/aXNBY/src/CxxWrap.jl:156

Stacktrace:
 [1] __cxxwrap_smartptr_construct_from_other(t::Type{CxxWrap.StdLib.SharedPtrDereferenced{Buffer}}, p::CxxWrap.StdLib.SharedPtrAllocated{Buffer})
   @ CxxWrap.CxxWrapCore ~/.julia/packages/CxxWrap/aXNBY/src/CxxWrap.jl:156
 [2] convert(#unused#::Type{CxxWrap.StdLib.SharedPtrDereferenced{Buffer}}, p::CxxWrap.StdLib.SharedPtrAllocated{Buffer})
   @ CxxWrap.CxxWrapCore ~/.julia/packages/CxxWrap/aXNBY/src/CxxWrap.jl:168
 [3] convert(#unused#::Type{CxxWrap.StdLib.SharedPtrDereferenced{Buffer}}, p::CxxWrap.StdLib.SharedPtrAllocated{LocalMemoryBuffer})
   @ CxxWrap.CxxWrapCore ~/.julia/packages/CxxWrap/aXNBY/src/CxxWrap.jl:175
 [4] cconvert(T::Type, x::CxxWrap.StdLib.SharedPtrAllocated{LocalMemoryBuffer})
   @ Base ./essentials.jl:492
 [5] RayObject(arg1::CxxWrap.CxxWrapCore.SmartPointer{T2} where T2<:Buffer, arg2::CxxWrap.CxxWrapCore.SmartPointer{T2} where T2<:Buffer, arg3::Union{CxxWrap.CxxWrapCore.ConstCxxRef{StdVector{ObjectReference}}, CxxWrap.CxxWrapCore.CxxRef{StdVector{ObjectReference}}, StdVector{ObjectReference}}, arg4::Union{Bool, CxxWrap.CxxWrapCore.CxxBool})
   @ ray_core_worker_julia_jll ~/.julia/packages/CxxWrap/aXNBY/src/CxxWrap.jl:624
 [6] top-level scope
   @ REPL[12]:1
```

Reply:
solved by dereferencing (?) the args in the constructor